### PR TITLE
OHRM5X-1082: Employee attendance get the top record API added

### PR DESCRIPTION
**Pull request by VageeshOrangeHRM**
This PR Contains following changes:
1. OHRM5X-1082 : End point added for the Attendance-PunchIn/Out-[API] to get latest attendance record for employee.
2. DAO and Integration test added.
3. Data group and role permission added for the endpoint.

### CODE REVIEW

File: installer/dbscript/dbscript-2.sql [hunk 1]
```diff
@@ -4381,14 +4381,19 @@ UPDATE `ohrm_user_role_data_group` SET `can_update` = 0 WHERE `data_group_id` =
 -- Attendance Module
 
 INSERT INTO ohrm_data_group (`name`, `description`, `can_read`, `can_create`,`can_update`, `can_delete`)
-VALUES ('apiv2_attendance_configuration', 'API-v2 Attendance - Attendance Configuration', 1, 0, 1, 0);
+VALUES ('apiv2_attendance_configuration', 'API-v2 Attendance - Attendance Configuration', 1, 0, 1, 0),
+       ('apiv2_employee_attendance_record', 'API-v2 Attendance - Employee Attendance', 1, 0, 0, 0);
```
Comment (VageeshOrangeHRM) [inline installer/dbscript/dbscript-2.sql:4385]: Needed to verify the naming convention of the Endpoint Class.

File: symfony/plugins/orangehrmAttendancePlugin/Api/EmployeeAttendanceAPI.php [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+use OrangeHRM\Core\Api\V2\ResourceEndpoint;
+use OrangeHRM\Core\Api\V2\Validator\ParamRule;
+use OrangeHRM\Core\Api\V2\Validator\ParamRuleCollection;
+use OrangeHRM\Core\Api\V2\Validator\Rule;
+use OrangeHRM\Core\Api\V2\Validator\Rules;
+use OrangeHRM\Core\Traits\Auth\AuthUserTrait;
+use OrangeHRM\Entity\AttendanceRecord;
+
+class EmployeeAttendanceAPI extends Endpoint implements ResourceEndpoint
+{
```
Comment (VageeshOrangeHRM) [inline symfony/plugins/orangehrmAttendancePlugin/Api/EmployeeAttendanceAPI.php:38]: Needed to verify the naming convention of the Endpoint Class.

File: symfony/plugins/orangehrmAttendancePlugin/config/routes.yaml [hunk 3]
```diff
@@ -16,16 +16,24 @@ apiv2_attendance_validate_datetime:
...
+#  controller: OrangeHRM\Attendance\Controller\AttendanceMockController::sendAttendanceResponse
+#  methods: [ GET, POST, PUT ]
 
 apiv2_attendance_timezones:
   path: /api/v2/attendance/timezones
   controller: OrangeHRM\Attendance\Controller\AttendanceMockController::getTimezones
   methods: [ GET ]
 
+apiv2_employee_attendance_record:
+  path: /api/v2/attendance/records/latest
```
Comment (VageeshOrangeHRM) [inline symfony/plugins/orangehrmAttendancePlugin/config/routes.yaml:30]: Needed to verify the naming convention of route tag name

File: phpunit.xml [hunk 4]
```diff
@@ -66,8 +66,8 @@
     </testsuite>
     <testsuite name="Attendance">
 <!--      <directory>symfony/plugins/orangehrmAttendancePlugin/test/Service</directory>-->
```
Comment (RajithaKumara) [inline phpunit.xml:68]: *

File: symfony/plugins/orangehrmAttendancePlugin/config/routes.yaml [hunk 5]
```diff
@@ -16,16 +16,24 @@ apiv2_attendance_validate_datetime:
   controller: OrangeHRM\Attendance\Controller\AttendanceMockController::validateDateTime
   methods: [ GET ]
 
-apiv2_attendance_records:
-  path: /api/v2/attendance/records/{id?}
-  controller: OrangeHRM\Attendance\Controller\AttendanceMockController::sendAttendanceResponse
-  methods: [ GET, POST, PUT ]
+#apiv2_attendance_records:
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmAttendancePlugin/config/routes.yaml:19]: *

**Review by RajithaKumara — APPROVED** [review]

**Pull request merged**